fendermint/vm/actor_interface/src/ipc.rs

```diff
@@ -145,10 +149,14 @@ impl ValidatorMerkleTree {
 pub mod gateway {
     use super::SubnetID;
     use ethers::contract::{EthAbiCodec, EthAbiType};
-    use ethers::core::types::{H160, U256};
+    use ethers::core::types::{Bytes, H160, U256};
     use fendermint_vm_genesis::ipc::GatewayParams;
+    use fendermint_vm_genesis::{Collateral, Validator};
+    use fvm_shared::address::Error as AddressError;
     use fvm_shared::address::Payload;
     use fvm_shared::econ::TokenAmount;
+    // TODO: This is what the contract _should_ take in the constructor.
+    use ipc_actors_abis::subnet_actor_getter_facet::GenesisValidator;
```

Comment:
I don't think I fully agree with this one. 
* When we bootstrap a child subnet with Fendermint, we should use the `IpcProvider` to get the genesis validators from the bootstrap  event (or the storage), This is a `GenesisValidator` type. 
* Fendermint uses this `GenesisValidator` to extract the public key and construct the gateway. I can pass a `GenesisValidator` as a construct parameter of the gateway if its easier for you, but we wouldn't be using the metadata and public keys for anything there.

Reply:
> When we bootstrap a child subnet with Fendermint, we should use the IpcProvider to get the genesis validators from the bootstrap event
That should happen in https://github.com/consensus-shipyard/fendermint/issues/306 to produce the JSON file. 

> I can pass a GenesisValidator as a construct parameter of the gateway if its easier for you, but we wouldn't be using the metadata and public keys for anything there.

You wouldn't be using it in the constructor, however we will need the public key again and again every time we create a new checkpoint and need to update the CometBFT power table in https://github.com/consensus-shipyard/fendermint/issues/252 

New joiners will have their public keys appear there, so it is only natural that _all_ validators should have it in the ledger. Otherwise I'd constantly have to go back to the parent to figure out whether some validators who don't have metadata might have been one of the genesis validators. 

The metadata and the public key are one and the same thing at the moment.